pkg/rancher-desktop/config/settingsImpl.ts

```diff
@@ -252,6 +261,34 @@ function fileIsWritable(path: string) {
   }
 }
 
+function migrateSettingsLocationOnWindows(): boolean {
+  if (migratedSettingsFileLocationOnWindows || process.platform !== 'win32') {
```

Comment:
I did it as a way of adding robustness to the following code:
```
function load() {
  try {
    return finishConfiguringSettings(loadFromDisk(), deploymentProfiles);
  } catch {

Reply:
> 4\. BUT `fs.copyFileSync(oldConfigPath, newConfigPath);` fails to actually copy the file, but doesn't throw an exception (due to some bug in the library)

This is the wrong approach to deal with this concern. If you think `fs.copyFileSync` can fail without throwing an error, you should call `fileExists(newConfigPath)` and thrown an error yourself if it still isn't there.

Deal with the issue locally instead of having a global variable tracking state.